### ISSUE TITLE
Weekly Community Check-In #5 - "Welcoming Newcomers Efficiently"

### ISSUE THREAD

**Issue by Rishabh570** [publiclab/plots2#4607]
Hi everyone :wave: !

We all at Public Lab :balloon: - learn, grow, work, brainstorm ideas, contribute together so why not share about our weekly goals and the awesome work we have done at Public Lab with each other, so we can support and collaborate with each other better. Let's start a Community Check-In from this week onwards, where every community member would share something about their work from the past week and about their current week's goal :dart:. You are also welcome to share fun-fact :smile: , new ideas :bulb: , your learning goals :ballot_box_with_check: .

If you're new here, welcome :balloon:, and please comment a Hello message below, we love to work with you. If you're looking for new issues, please try some of our first-timers-only issues.

We're SO EXCITED to have your help!

Is there anything, you would like to share with us from past week's work? What is your plan for this week?

If you have not planned yet, just leave a Hello! so that we know that you are in sync with us :arrows_clockwise: and doing well!

As always, if you're waiting for a review, or if you're stuck, please request help here OR leave a comment with @publiclab/mentors @publiclab/reviewers for some input. :raised_hands:

### This Week's Theme

I thought of making this week's theme around the idea of `Welcoming Newcomers Efficiently`.
As we have [code.publiclab.org](https://code.publiclab.org) which provides a nice list of issues which can be solved by the newcomers to get up and running with the project, merging those PRs takes time sometimes...I think discussing the various ways which can improve and speed-up this process would enhance newcomer's experience to much greater extent.

If you have any suggestion or query, feel free to comment down below. 

Have a great week everyone :tada: :fireworks: :balloon: 

cc
@thesparks @faithngetich @DakshGondaliya @mohitRJranjan @milaaraujo @kevinzluo
@stefannibrasil @gauravano @ViditChitkara @tech4GT @mridulnagpal @namangupta01 @sagarpreet-chadha @Souravirus @jywarren @MayankKashyap @siaw23 @ryzokuken @ccpandhare @icarito @steviepubliclab @ebarry @jywarren, @sagarpreet-chadha, @icarito, @JonathanXu1, @uzorjchibuzor, @eli6, @kevinzluo, @ValentinaTironi, @rexagod, @sashadev-sky, @IshaGupta18, @milaaraujo, @dinaelhanan, @cesswairimu, @dependabot[bot], @oorjitchowdhary, @okonek, @wanzulfikri, @mohitRJranjan @publiclab/mentors @publiclab/image-sequencer-guides @publiclab/leaflet-environmental-layers-guides @Paarmita @vishalka98 @publiclab/reviewers @Divy123 @pomyo @bhavayAnand9 @ovipa017-xamk @Mridul97 @avsingh999 @Rishabh570 @subhahu123 @aSquare14 @Radhikadua123 @quinn-codes-synthesis @amychan331 @mardelvalle @narnt @hydro-m @love-opensource @romanrodriguez @shubhangi-chauhan @Dhiraj240 @geekychaser @codeIriss @CoderJolly 

### Summer of Code
@jywarren has started a brainstorming post for the ideas page for Summer of Code 2019 -It would be best get some ideas into discussion from the fellow community member before time. People have been brainstorming already - let's do it together!
https://publiclab.org/notes/warren/01-02-2019/brainstorming-for-summer-of-code-2019

### Software Contributors Survey
@ebarry reminded us about a survey that was conducted the year before and we are willing to hear it from you again. The idea discussed in 2017 is here: https://publiclab.org/questions/warren/10-25-2017/help-with-a-standard-mini-evaluation-for-assessing-software-outreach-efforts

Here's the survey as sent out on December 14, 2017: #1890 so, it would great to run the survey this time too.

### People who did the check-ins

- [x] Gaurav Sachdeva (@gauravano )
- [x] Sidharth Bansal (@SidharthBansal  )
- [x] Cess (@cesswairimu )
- [x] Harman Jolly (@CoderJolly )
- [x] Rishabh Rawat (@Rishabh570 )

**Comment by SidharthBansal** [comment]
Hello everyone,
I will be working on Multiple Subscription Prompts across PL this week. We want to increase our community so I am dividing my projects into small small pieces to integrate up whole projects. We will be creating `first timers only` issue for the whole project very soon. I will abide by this week's title @Rishabh570 
Login Sequence Project is almost completed. 26 issues/prs are left. Else 175 issues/prs closed and merged. 
I will like to thank all the contributors at PL for taking up bigger projects and trying to expand our community.

@Rishabh570 it will be great if you can add a checklist of all the people you did the checkins and edit the issue.
Who will be willing to do the next weekly checkin? Please tell your name asap.

**Comment by Rishabh570** [comment]
@SidharthBansal Updated.

**Comment by SidharthBansal** [comment]
Thanks

**Comment by ebarry** [comment]
Hi all, the Software Contributors Survey is waiting on just one final nod from another team, and then i will post it as a Github issue for everyone here to fill out! Expecting this to happen later today. Thanks and stay tuned!

**Comment by oorjitchowdhary** [comment]
Hi all, this week's theme being `welcoming newcomers`, I'll try to make as many ftos from #4581.
Also, I've another set of exams this week so I won't be online for a long time. 😞

**Comment by geekychaser** [comment]
@oorjitchowdhary No problem i will try making more ftos and All the best for your exams :smile:

**Comment by SidharthBansal** [comment]
Hi all we have one more issue which needs to be broken down into ftos. If
anyone wants to work with me, then it will be awesome. Issue no #4131.
Please comment on this issue. I will tell you how to work in which
direction.
Thanks all.
Oorjit we will love to see you back soon.

**Comment by geekychaser** [comment]
@SidharthBansal Yes i would love to work on this.
thank you

**Comment by SidharthBansal** [comment]
We will start on that issue after one two related PRS are merged.
Let's make pl most friendly learning environment + collaborative + hard
working organisation in the world. You all are simply awesome!!!!!!!!!

**Comment by rexagod** [comment]
In context of this week's check-in theme, I opened up three new `fto`s today at [Editor's 232](https://github.com/publiclab/PublicLab.Editor/issues/232), [Editor's 231](https://github.com/publiclab/PublicLab.Editor/issues/231), and [Image-Sequencer's 672](https://github.com/publiclab/image-sequencer/issues/672). Kudos to the PublicLab community and everyone constantly working hard to open new `fto`s every single day! I'm currently occupied by a [long standing issue](https://github.com/publiclab/PublicLab.Editor/issues/225) but nonetheless will try to open up even more first timers very soon.

Thanks everyone for making this possible, let's do this! :tada:

**Comment by oorjitchowdhary** [comment]
Also, just a suggestion.. I created the 'good first issue' label today for plots2.. 
Let's start 'good first issue' label from this week as GitHub has a special feature for 'good first issue' and 'help wanted' labels.. 
These let people discover repos..

**Comment by gauravano** [comment]
Thanks @oorjitchowdhary for working with labels. Also, I would suggest reviewers to assign a contributor to issue whenever someone claims it and remove `help wanted` label too for that time, so that newcomers can find un-assigned issues easily. 

**Week's plan**
1. I would work on Translation project this week - https://github.com/publiclab/plots2/projects/9
2. Would work on solving the tasks I assigned myself on plots2 
3. Reviewing and merging PR's
4. Completing all Public Lab's issue templates/bot installation/docs - I will try to do these tasks by opening FTO's 

Thanks and wishing you all a wonderful week!

**Comment by jywarren** [comment]
This sounds good @oorjitchowdhary -- just let's not forget to also use `first-timers-only` as it means something even more specific. Great!!

**Comment by jywarren** [comment]
Liz posted the community survey here today: https://github.com/publiclab/plots2/issues/4614 🎉 

@Rishabh570 would you mind putting it somewhere prominent? 

Shall we add it temporarily to the Welcomebot configs? I'll pin it too.

**Comment by gauravano** [comment]
Yes, adding it to welcome bot would surely give more attention to the survey.

**Comment by Rishabh570** [comment]
@jywarren Welcome bot would surely give it more exposure...can we add it temporarily to `code.publiclab.org`? That can help too I think.

**Comment by cesswairimu** [comment]
Hey everyone!
* In the spirit of welcoming newcomers I broke up some issues in my project but they are still not good ftos so I will be breaking them further and adding more step-by-step info on them later today.
* Just a suggestion, maybe we can have an issue for installations where anyone stuck with installing and local setup can ask for help and other newcomers can refer incase they get a similar issue. The installation docs are awesome but once in a while someone asks for help on gitter or riot and the conversation is hard to track. It may be a good place to also decide when we should update the docs too :thinking:.
* This week I will be finishing up on implementing new graphs on the stats page. A lot of Javascript. Anyone comfortable with JS that I can reach out to if I get stuck here?
Great theme @Rishabh570 :tada:, 
Nice week all!

**Comment by rexagod** [comment]
Hello @cesswairimu! Let me know if you need any help in JS, I'll be happy to help! :smile:

**Comment by dewanhimanshu** [comment]
Hello everyone .I would like to contribute to community.

**Comment by cesswairimu** [comment]
Thanks @rexagod will do

**Comment by gauravano** [comment]
@cesswairimu we already have a issue for discussing installation queries - https://github.com/publiclab/plots2/issues/3840

**Comment by rexagod** [comment]
Hi @dewanhimanshu! You can go ahead and start contributing by picking one of the [`first-timers-only`](https://github.com/publiclab/plots2/issues?q=is%3Aopen+is%3Aissue+label%3Afirst-timers-only) issues of the repository.

Welcome to the Public Lab community. Cheers!

**Comment by Rishabh570** [comment]
Hi all, I'll start working towards community-toolbox MVC architecture now...I'll try to break it into different milestones as it is a big change and I'll also try to create some `fto`s along the way.

Also, I've opened up [this](https://github.com/publiclab/community-toolbox/issues/131) issue to discuss what features/enhancements we need in `code.publiclab.org`...please comment there if anyone wants to suggest anything regarding that.

Thanks!!! :raised_hands: :tada:

**Comment by jywarren** [comment]
Hi, everyone! Great to see you all!

@dinaelhanan helped write an excellent test that clearly demonstrates the bug @ebarry reported here: https://github.com/publiclab/plots2/issues/4121 -- now it's ready for a new contributor to problem-solve. 

This one is cool because it's now test-driven development (TDD) -- we have captured the bug with a test, now any solution that fixes that test should be good! Anyone want to take a look at it? It's related to search results using the search API.

**Comment by jywarren** [comment]
If anyone's interested in a CSS-centric issue, we've been having some trouble getting these images to be restricted to a fixed height and width, but using `scroll:hidden` the way images work [on the dashboard](https://publiclab.org/research): https://publiclab.org/wiki/micro#Events

Lots of context here: https://github.com/publiclab/plots2/pull/4198 and a new issue here: https://github.com/publiclab/plots2/issues/4524

We'd love some help!

**Comment by rexagod** [comment]
Hi all! Following @jywarren's [recommendation](https://github.com/publiclab/PublicLab.Editor/issues/227#issuecomment-454133088), I wanted to know the community's thoughts and opinions on shifting the current task runner over at the Editor from `grunt` to `gulp` or `webpack`. Although @geekychaser, @himanshuc3, and I are a little inclined towards using `webpack.js`, we would love to hear from you about any pros/cons that may come to your mind regarding the same.

Thanks!

**Comment by dhruvgupta167** [comment]
Hello Everyone! I have already solved one FTO issue and opened a few issues as well. This week I'll try to contribute to at least one help-wanted issue.

**Comment by dhruvgupta167** [comment]
Hello everyone! I have already solved one FTO issue, thanks to @SidharthBansal . This week I'll try to contribute to at least one help wanted issue.

**Comment by sashadev-sky** [comment]
@SidharthBansal When would you like me to make the post? I am on EST but most people aren't so I want to plan accordingly. 

Do I choose my own theme or did you have one in mind?

**Comment by SidharthBansal** [comment]
You can post on your Sunday at any time. Don't consider time zones.
Thanks for thinking deeply. I appreciate it.
Any theme will work.

**Comment by CoderJolly** [comment]
Who would be opening the next Community Check-In ?

**Comment by geekychaser** [comment]
Created several `first-timer-issue` for welcoming newcomers #4673 #4663 #4666 #4662 
Thank you

**Comment by gauravano** [comment]
Great work @geekychaser :100:

**Comment by SidharthBansal** [comment]
Moved to #4676
